lib/timers.js

```diff
@@ -132,6 +147,22 @@ exports._unrefActive = function(item) {
 };
 
 
+function timerEmitBefore(asyncId, triggerId) {
+  if (async_hook_fields[kBefore] > 0)
+    emitBefore(asyncId, triggerId);
+  else
+    pushAsyncIds(asyncId, triggerId);
+}
+
+
+function timerEmitAfter(asyncId) {
+  if (async_hook_fields[kAfter] > 0)
+    emitAfter(asyncId);
+  else
+    popAsyncIds(asyncId);
+}
```

Comment:
The two locations that are the same are here and in `lib/internal/process/next_tick.js`. Could be exported into a single function. Will do some benchmarking (have seen performance issues in the past when running functions across modules). Also I'm doing everything I can to avoid exporting any underscore methods. Will think how I could export this while avoiding an underscore fn.